src/Schema/Column/AbstractColumnFactory.php

```diff
@@ -138,13 +182,14 @@ protected function columnDefinitionParser(): ColumnDefinitionParser
 
     /**
      * @psalm-param ColumnType::* $type
-     * @param ColumnInfo $info
-     *
+     * @psalm-param ColumnInfo $info
+     * @psalm-assert ColumnInfo $info
      * @psalm-return class-string<ColumnInterface>
      */
-    protected function getColumnClass(string $type, array $info = []): string
+    protected function getColumnClass(string $type, array &$info = []): string
     {
-        return match ($type) {
+        /** @psalm-var class-string<ColumnInterface> */
+        return $this->mapType($this->columnClassMap, $type, $info) ?? match ($type) {
```

Comment:
Extract `match ($type) { ...` to separate method. This allow to prevent duplicate `$this->mapType(...)` in `yiisoft/db-*` packages.